### PR TITLE
Ticket28076 rmlink or header

### PR DESCRIPTION
**Pull request by juga0**

### CODE REVIEW

File: sbws/core/generate.py [hunk 1]
```diff
@@ -61,6 +61,9 @@ def gen_parser(sub):
                         "other.")
     p.add_argument('-n', '--min-num', default=NUM_MIN_RESULTS, type=int,
                    help="Mininum number of a results to consider them.")
+    p.add_argument('-l', '--rm-link', action='store_true',
+                   help='If specified, remove latest.v3bw link when there '
+                        'are no relays to include in it.')
```
Comment (teor2345) [inline sbws/core/generate.py]: Does sbws need to be able to remove the link?
It seems simpler and more reliable to always write the file, and always have the link.

File: sbws/lib/v3bwfile.py [hunk 2]
```diff
@@ -887,10 +895,13 @@ def rm_link(self, output):
             except FileNotFoundError:
                 pass
 
-    def write(self, output):
+    def write(self, output, rm_link=False):
         if output == '/dev/stdout':
             log.info("Writing to stdout is not supported.")
             return
+        if not self.is_min_perc and rm_link:
```
Comment (teor2345) [inline sbws/lib/v3bwfile.py]: Does sbws need to be able to remove the link?
It seems simpler and more reliable to always write the file, and always have the link.

File: sbws/core/generate.py [hunk 3]
```diff
@@ -95,21 +95,18 @@ def main(args, conf):
...
     output = args.output or \
         conf.getpath('paths', 'v3bw_fname').format(now_fname())
-    # Check percentage of measured relays.
-    consensus_path = os.path.join(conf.getpath('tor', 'datadir'),
-                                  "cached-consensus")
-    if is_min_percent_measured(consensus_path):
-        bw_file.write(output)
-    else:
-        bw_file.rm_link(output)
+    bw_file.write(output, args.rm_link)
```
Comment (teor2345) [inline sbws/core/generate.py]: Does sbws need to be able to remove the link?
It seems simpler and more reliable to always write the file, and always have the link.

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,6 +17,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Number of results comparison and number of results away from each other are
   incorrect (#28041)
 
+## Added
+
+- Write bw file only when the percentage of measured relays is bigger than 60%
```
Comment (teor2345) [inline CHANGELOG.md]: This branch needs an updated changelog entry.

File: sbws/lib/v3bwfile.py [hunk 5]
```diff
@@ -842,6 +842,17 @@ def to_plt(self, attrs=['bw'], sorted_by=None):
         ys = [[getattr(l, k) for l in self.bw_lines] for k in attrs]
         return x, ys, attrs
 
+    def rm_link(self, output):
```
Comment (teor2345) [inline sbws/lib/v3bwfile.py]: Does sbws need to be able to remove the link?
It seems simpler and more reliable to always write the file, and always have the link.

**Comment by juga0** [comment]
Merged

**Pull request closed without merging**